### DISCUSSION TITLE
Proposal: Migrate documentation to serverless cloud platform deployment

### DISCUSSION THREAD

**Discussion by misitebao** [go-task/task#1193] [General]
Hi, @andreynering @pd93 

Currently our website is deployed on `github-pages`, relying on `website-deploy` and `website-test` workflow it works well. The problem now is that we have no way to preview when there are commits or PRs about documentation changes.

For myself, I deploy the `fork` project in `Vercel`, so you can see that basically the PR I submit about the documentation will come with a preview link, but we can fully automate and popularize this method change.

We can deploy our documents on the cloud platform. Every time we change the document, we can check the status and preview link of the document deployment of the specified branch in real time, which will provide great convenience.

As far as I know and have used cloud platforms, there are `Vercel`, `Netlify`, `Cloudflare Pages`, and the usage methods are basically the same. Since Task is an organization rather than a personal project, `Vercel` is not recommended.

**Comment by andreynering** [comment]
I have a bit of experience with Netlify, although I just deploy via "drag & drop" most of the time. 🙂 

They have a [free plan for Open Source](https://www.netlify.com/legal/open-source-policy/), but which requires a link or badge to be added to our website:

> Must feature a link to our service on your main page, or all internal pages. You have two options:
> - We have [premade badges](https://www.netlify.com/press/#badges) for your convenience, or
> - You may create your own link, which should read “This site is powered by Netlify”, and include a link back to our home page.

I don't have a strong opinion on this one. I honestly think GitHub Pages is good enough. Preview links on PRs would be a nice to have, but also not something we couldn't live without.

Maybe it'd be worth the change if the setup is simple enough. I never tried to integrate it with GitHub before to know.

**Comment by pd93** [comment]
Hi @misitebao, no strong opinion on this one. I usually checkout PRs on my machine to test them anyway, so if there are docs changes, I will just run `task docs` and take a look at any changes locally. It's definitely not the most seamless way to do it, so I'm open to improvements, but not something that's super important to me.

**Comment by misitebao** [comment]
Whether to migrate has no effect on me, I just offer a small suggestion. 😂

**Comment by andreynering** [comment]
@misitebao I end up applying for the OSS Netlify plan and we got accepted. The setup was quickier than I thought initially.

It's already generating preview links on PRs. I'll add a Netlify link/badge and move the domain to Netlify soon.
